src/Admin/Exports/Writer/CsvExportWriter.php

```diff
@@ -0,0 +1,151 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Admin\Exports\Writer;
+
+use RuntimeException;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class CsvExportWriter
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Admin\Exports\Writer
+ */
+class CsvExportWriter {
+	/**
+	 * Name of the subdirectory inside uploads/ to store exports.
+	 */
+	protected const EXPORT_FOLDER = 'gla-exports';
+
+	/**
+	 * Filesystem handler.
+	 *
+	 * @var WP_Filesystem_Direct
+	 */
+	protected $fs;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		global $wp_filesystem;
+
+		if ( ! $wp_filesystem ) {
+			require_once ABSPATH . '/wp-admin/includes/file.php';
+			WP_Filesystem();
+		}
+
+		$this->fs = $wp_filesystem;
+	}
+
+	/**
+	 * Create an export file.
+	 *
+	 * @param string $filename
+	 *
+	 * @return string
+	 *
+	 * @throws RuntimeException When unable to create a directory or file.
+	 */
+	public function create_file( string $filename ): string {
+		$upload_dir = wp_upload_dir();
+
+		if ( empty( $upload_dir['basedir'] ) || ! is_dir( $upload_dir['basedir'] ) ) {
+			// phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
+			throw new RuntimeException( 'Unable to determine upload directory.' );
+		}
+
+		$dir_path = trailingslashit( $upload_dir['basedir'] ) . self::EXPORT_FOLDER;
+
+		if ( ! $this->fs->is_dir( $dir_path ) ) {
+			wp_mkdir_p( $dir_path );
+
+			if ( ! $this->fs->is_dir( $dir_path ) ) {
+				// phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
+				throw new RuntimeException( 'Failed to create export directory: ' . $dir_path );
```

Comment:
This one likely should be escaped since the `$dir_path` could be a filtered value. Low risk, but still...

```suggestion
				throw new RuntimeException( sprintf( 'Failed to create export directory: %s', esc_html( $dir_path ) );
```